tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -803,3 +804,365 @@ def test_csrlsvqr(self, dtype):
+                                     xp.asarray(analytic_w[-p:]))
+
+        fiedler_guess = numpy.concatenate((numpy.ones(n//2),
+                                           -numpy.ones(n-n//2)))
+        X = numpy.vstack((numpy.ones(n), fiedler_guess)).T
+        lobpcg_w, _ = sp.linalg.lobpcg(xp.asarray(L), xp.asarray(X),
+                                       largest=False)
+        # Mathematically, the smaller eigenvalue should be zero
+        # and the larger should be the algebraic connectivity.
+        lobpcg_w = xp.sort(lobpcg_w)
+        eval_list.append(lobpcg_w)
+        cupy.testing.assert_allclose(lobpcg_w, xp.asarray(analytic_w[:2]),
+                                     atol=1e-14)
+        return eval_list
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_fiedler_small_8(self, xp, sp):
+        """Check the dense workaround path for small matrices.
+        """
+        # This triggers the dense path because 8 < 2*5.
+        return self._check_fiedler(8, 2, xp, sp)
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_fiedler_large_12(self, xp, sp):
+        """Check the dense workaround path avoided for non-small matrices.
+        """
+        # This does not trigger the dense path, because 2*5 <= 12.
+        return self._check_fiedler(12, 2, xp, sp)
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_eigs_consistency(self, xp, sp):
+        """Check eigs vs. lobpcg consistency.
+        """
+        # The n=5 case tests the alternative small matrix code path
+        param_arr = [(20, 1e-3), (5, 1e-8)]
+        eval_list = []
+        for n, _atol in param_arr:
+            vals = xp.arange(1, n+1, dtype=xp.float64)
+            A = sp.spdiags(vals, 0, n, n)
+            numpy.random.seed(345678)
+            X = xp.asarray(numpy.random.rand(n, 2))
+            lvals, lvecs = sp.linalg.lobpcg(A, X, largest=True, maxiter=100)
+            eval_list.append(lvals)
+            try:  # cupyx.sparse.linalg.eigsh might not converge!
+                vals, _ = sp.linalg.eigsh(A, k=2)
+            except Exception as e:
+                print("Exception: {} \n".format(e))
+                continue
+            self._check_eigen(A, lvals, lvecs, xp, sp, atol=_atol, rtol=0)
+            cupy.testing.assert_allclose(xp.sort(vals), xp.sort(lvals),
+                                         atol=1e-14)
+        return eval_list
+
+    def test_verbosity(self):
+        """Check that nonzero verbosity level code runs.
+        """
+        A, B = self._elasticRod(100, cupy)
+        n = A.shape[0]
+        m = 20
+        cupy.random.seed(0)
+        V = numpy.random.rand(n, m)
+        X = scipy.linalg.orth(V)
+        _, _ = sparse.linalg.lobpcg(A, cupy.asarray(X), B=B, tol=1e-5,
+                                    maxiter=30, largest=False,
+                                    verbosityLevel=9)
```

Comment:
Could you check the verbose message follows the same format as SciPy's one?

Reply:
I am yet to figure out a way to capture its stdout output in the tests fixtures for comparison. Do you have any suggestions?

Reply:
How about using [`contextlib.redirect_stdout`](https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stdout)?

Reply:
thanks @asi1024 , have tried to format the output for testing purposes. 👍🏾